src/core/compact_object.h

```diff
@@ -147,12 +147,14 @@ class CompactObj {
 
    private:
     friend class CompactObj;
-    explicit StrEncoding(uint8_t enc) : enc_(static_cast<EncodingEnum>(enc)) {
+    explicit StrEncoding(uint8_t enc, bool is_key)
+        : enc_(static_cast<EncodingEnum>(enc)), is_key_(is_key) {
     }
 
     size_t DecodedSize(size_t compr_size, uint8_t first_byte) const;
 
     EncodingEnum enc_;
+    bool is_key_;
```

Comment:
The alternative is keeping two different enum values for huffman encoding - like we do with ascii pack, depending on the dat size. Maybe it's cleaner this way?